homeassistant/components/aidot/light.py

```diff
@@ -0,0 +1,273 @@
+        self, hass: HomeAssistant, device, coordinator: AidotCoordinator
+    ) -> None:
+        """Initialize the light."""
+        super().__init__(coordinator, {device[CONF_ID]})
+        self.device = device
+        self.user_info = coordinator.client.login_info
+        self._attr_unique_id = device[CONF_ID]
+        self._attr_name = device[CONF_NAME]
+        self.pingtask = None
+        self.recvtask = None
+
+        modelId = device[CONF_MODEL_ID]
+        manufacturer = modelId.split(".")[0]
+        model = modelId[len(manufacturer) + 1 :]
+        mac = format_mac(device[CONF_MAC]) if device[CONF_MAC] is not None else ""
+        identifiers: set[tuple[str, str]] = (
+            set({(DOMAIN, self._attr_unique_id)}) if self._attr_unique_id else set()
+        )
+        self._attr_device_info = DeviceInfo(
+            identifiers=identifiers,
+            connections={(CONNECTION_NETWORK_MAC, mac)},
+            manufacturer=manufacturer,
+            model=model,
+            name=device[CONF_NAME],
+            hw_version=device[CONF_HARDWARE_VERSION],
+        )
+        self._cct_min = 0
+        self._cct_max = 0
+
+        supported_color_modes = set()
+        if CONF_PRODUCT in device and CONF_SERVICE_MODULES in device[CONF_PRODUCT]:
+            for service in device[CONF_PRODUCT][CONF_SERVICE_MODULES]:
+                if service[CONF_IDENTITY] == Identity.RGBW:
+                    supported_color_modes.add(ColorMode.RGBW)
+                elif service[CONF_IDENTITY] == Identity.CCT:
+                    self._cct_min = int(service[CONF_PROPERTIES][0][CONF_MINVALUE])
+                    self._cct_max = int(service[CONF_PROPERTIES][0][CONF_MAXVALUE])
+                    supported_color_modes.add(ColorMode.COLOR_TEMP)
+
+        if ColorMode.RGBW in supported_color_modes:
+            self._attr_color_mode = ColorMode.RGBW
+            self._attr_supported_color_modes = {ColorMode.RGBW, ColorMode.COLOR_TEMP}
+        elif ColorMode.COLOR_TEMP in supported_color_modes:
+            self._attr_color_mode = ColorMode.COLOR_TEMP
+            self._attr_supported_color_modes = {ColorMode.COLOR_TEMP}
+        else:
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+
+        self.lanCtrl = Lan(device, self.user_info)
+        self.lanCtrl.setUpdateDeviceCb(self.updateState)
+
+    async def _cleanup_resources(self):
+        """Release task."""
+        self.lanCtrl.setUpdateDeviceCb(None)
+        if hasattr(self, "pingtask") and self.pingtask is not None:
+            self.pingtask.cancel()
+        if hasattr(self, "recvtask") and self.recvtask is not None:
+            self.recvtask.cancel()
+        await self.lanCtrl.reset()
+
+    async def _try_connect(self):
+        """Try connect."""
+        aidot_coordinator: AidotCoordinator = self.coordinator
+        device_ip = aidot_coordinator.discovered_devices.get(self.device[CONF_ID])
```

Comment:
```suggestion
        device_ip = self.coordinator.discovered_devices.get(self.device[CONF_ID])
```
should be correctly typed now